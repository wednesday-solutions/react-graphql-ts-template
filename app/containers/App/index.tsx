/**
 *
 * App.js
 *
 * This component is the skeleton around the actual pages, and should only
 * contain code that should be seen on all pages. (e.g. navigation bar)
 *
 */
import React from 'react';
import globalStyle from '@app/global-styles';
import { routeConfig } from '@app/routeConfig';
import { Global } from '@emotion/react';
import map from 'lodash-es/map';
import { withRouter } from 'react-router';
import { Route, Switch } from 'react-router-dom';
import { compose } from 'redux';
import { ThemeProvider as MUIThemeProvider, createTheme, StyledEngineProvider } from '@mui/material/styles';
import { CssBaseline, Grid } from '@mui/material';
import For from '@components/For';
import Header from '@components/Header';
import { colors } from '@themes/index';
import Sidebar from '@app/components/Siderbar';
import { SCREEN_BREAK_POINTS, HEADER_HEIGHT, MIN_SIDEBAR_WIDTH } from '@app/utils/constants';

declare module '@mui/material/styles' {
  export interface Theme {
    baseLayout: {
      headerHeight: string;
      sidebarWidth: string;
    };
  }
  // allow configuration using `createTheme`
  export interface ThemeOptions {
    baseLayout?: {
      headerHeight: string;
      sidebarWidth: string;
    };
  }
}

export const theme = createTheme({
  palette: {
    primary: {
      main: colors.primary
    },
    secondary: {
      main: colors.secondary
    }
  },
  baseLayout: {
    headerHeight: HEADER_HEIGHT,
    sidebarWidth: MIN_SIDEBAR_WIDTH
  },
  breakpoints: {
    values: {
      xs: 0,
      sm: SCREEN_BREAK_POINTS.mobile,
      md: SCREEN_BREAK_POINTS.tablet,
      lg: SCREEN_BREAK_POINTS.desktop,
      xl: SCREEN_BREAK_POINTS.largeDesktop
    }
  }
});

export function App() {
  return (
    <StyledEngineProvider injectFirst>
      <MUIThemeProvider theme={theme}>
        <CssBaseline />
        <Global styles={globalStyle} />
        <Grid container>
          <Grid item xs={12}>
            <Header />
          </Grid>
          <Grid item xs={0.5}>
            <Sidebar />
          </Grid>
          <Grid item xs={11.5}>
            <For
              ParentComponent={(props) => <Switch {...props} />}
              of={map(Object.keys(routeConfig))}
              renderItem={(routeKey, index) => {
                const Component = routeConfig[routeKey].component;
                return (
                  <Route
                    exact={routeConfig[routeKey].exact}
                    key={index}
                    path={routeConfig[routeKey].route}
                    render={(props) => {
                      const updatedProps = {
                        ...props,
                        ...routeConfig[routeKey].props
                      };
                      return <Component {...updatedProps} />;
                    }}
                  />
                );
              }}
            />
          </Grid>
        </Grid>
      </MUIThemeProvider>
    </StyledEngineProvider>
  );
}

export default compose(withRouter)(App);
