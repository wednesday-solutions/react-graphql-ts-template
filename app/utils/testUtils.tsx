import React from 'react';
import { IntlProvider } from 'react-intl';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { Route, Router } from 'react-router-dom';
import { ThemeProvider } from '@mui/material/styles';
import configureStore from '@app/configureStore';
import { DEFAULT_LOCALE, translationMessages } from '@app/i18n';
import ConnectedLanguageProvider from '@containers/LanguageProvider';
import IntlGlobalProvider from '@components/IntlGlobalProvider';
import history from './history';
import { theme } from '@containers/App';

export const renderWithIntl = (children: React.ReactNode) =>
  render(
    <IntlProvider locale={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
      <IntlGlobalProvider>{children}</IntlGlobalProvider>
    </IntlProvider>
  );

export const renderProvider = (children: React.ReactNode, { path }: { path?: string } = {}, renderFn = render) => {
  const store = configureStore({}).store;
  return renderFn(
    <Provider store={store}>
      <ConnectedLanguageProvider messages={translationMessages}>
        <ThemeProvider theme={theme}>
          <Router history={history}>{path ? <Route path={path}>{children}</Route> : children}</Router>
        </ThemeProvider>
      </ConnectedLanguageProvider>
    </Provider>
  );
};
export const timeout = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
export const apiResponseGenerator = <Data,>(ok: boolean, data: Data, error?: object) => ({
  ok,
  data,
  error
});

export const getStyles = () => {};
