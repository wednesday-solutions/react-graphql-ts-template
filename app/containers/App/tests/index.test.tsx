import React from 'react';
import { renderProvider } from '@utils/testUtils';
import App from '@containers/App';
import { BrowserRouter } from 'react-router-dom';
import { waitFor } from '@testing-library/react';

describe('<App /> container tests', () => {
  it('should render and match the snapshot', async () => {
    const { container } = renderProvider(
      <BrowserRouter>
        <App />
      </BrowserRouter>
    );
    await waitFor(() => expect(container.textContent).toContain('Wednesday Solutions'));
    await waitFor(() => expect(container).toMatchSnapshot());
  });
});
