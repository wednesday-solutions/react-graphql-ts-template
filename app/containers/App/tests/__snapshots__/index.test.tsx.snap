// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> container tests should render and match the snapshot 1`] = `
.emotion-0 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-1 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 100%;
  -ms-flex-preferred-size: 100%;
  flex-basis: 100%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 100%;
}

@media (min-width:320px) {
  .emotion-1 {
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 100%;
  }
}

@media (min-width:768px) {
  .emotion-1 {
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 100%;
  }
}

@media (min-width:992px) {
  .emotion-1 {
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 100%;
  }
}

@media (min-width:1400px) {
  .emotion-1 {
    -webkit-flex-basis: 100%;
    -ms-flex-preferred-size: 100%;
    flex-basis: 100%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 100%;
  }
}

.emotion-2.emotion-2 {
  padding: 0 1rem;
  height: 7rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  background-color: #000000;
  gap: 1rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-4 {
  height: 5rem;
  width: auto;
  object-fit: contain;
}

@media (max-width:319.95px) {
  .emotion-4 {
    height: 4rem;
  }
}

.emotion-7.emotion-7 {
  font-size: 1.5rem;
  font-weight: bold;
}

.emotion-7.emotion-7 {
  margin: 0;
  font-size: 1.125rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  color: #FFFFFF;
}

@media (min-width: 48em) {
  .emotion-7.emotion-7 {
    font-size: 1.5rem;
  }
}

@media (min-width: 62em) {
  .emotion-7.emotion-7 {
    font-size: 2rem;
  }
}

.emotion-9 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 4.166667%;
  -ms-flex-preferred-size: 4.166667%;
  flex-basis: 4.166667%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 4.166667%;
}

@media (min-width:320px) {
  .emotion-9 {
    -webkit-flex-basis: 4.166667%;
    -ms-flex-preferred-size: 4.166667%;
    flex-basis: 4.166667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 4.166667%;
  }
}

@media (min-width:768px) {
  .emotion-9 {
    -webkit-flex-basis: 4.166667%;
    -ms-flex-preferred-size: 4.166667%;
    flex-basis: 4.166667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 4.166667%;
  }
}

@media (min-width:992px) {
  .emotion-9 {
    -webkit-flex-basis: 4.166667%;
    -ms-flex-preferred-size: 4.166667%;
    flex-basis: 4.166667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 4.166667%;
  }
}

@media (min-width:1400px) {
  .emotion-9 {
    -webkit-flex-basis: 4.166667%;
    -ms-flex-preferred-size: 4.166667%;
    flex-basis: 4.166667%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 4.166667%;
  }
}

.emotion-10 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-12.emotion-12 {
  width: 6%;
  min-width: 4.5rem;
  max-width: 7rem;
  min-height: calc(100vh - 7rem);
  height: auto;
  background-color: #000000;
  display: inline;
  text-align: center;
}

.emotion-14.emotion-14 {
  margin-top: 1rem;
  object-fit: contain;
}

.emotion-16 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-basis: 95.833333%;
  -ms-flex-preferred-size: 95.833333%;
  flex-basis: 95.833333%;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  max-width: 95.833333%;
}

@media (min-width:320px) {
  .emotion-16 {
    -webkit-flex-basis: 95.833333%;
    -ms-flex-preferred-size: 95.833333%;
    flex-basis: 95.833333%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 95.833333%;
  }
}

@media (min-width:768px) {
  .emotion-16 {
    -webkit-flex-basis: 95.833333%;
    -ms-flex-preferred-size: 95.833333%;
    flex-basis: 95.833333%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 95.833333%;
  }
}

@media (min-width:992px) {
  .emotion-16 {
    -webkit-flex-basis: 95.833333%;
    -ms-flex-preferred-size: 95.833333%;
    flex-basis: 95.833333%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 95.833333%;
  }
}

@media (min-width:1400px) {
  .emotion-16 {
    -webkit-flex-basis: 95.833333%;
    -ms-flex-preferred-size: 95.833333%;
    flex-basis: 95.833333%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 95.833333%;
  }
}

<div>
  <div
    class="MuiGrid-root MuiGrid-container emotion-0"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 emotion-1"
    >
      <div
        class="emotion-2 emotion-3"
        data-testid="header"
      >
        <a
          href="/"
        >
          <img
            alt="logo"
            class="emotion-4 emotion-5"
            src="IMAGE_MOCK"
          />
        </a>
        <p
          class="emotion-6 emotion-7 emotion-8"
          data-testid="t"
        >
          Wednesday Solutions
        </p>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-0.5 emotion-9"
    >
      <div
        class="emotion-10 emotion-11"
      >
        <div
          class="emotion-12 emotion-13"
          data-testid="sidebar"
        >
          <a
            aria-label="home link"
            data-testid="rocket-home-link"
            href="/"
          >
            <img
              alt="rocket-icon"
              class="emotion-14 emotion-15"
              src="IMAGE_MOCK"
            />
          </a>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-11.5 emotion-16"
    />
  </div>
</div>
`;
